nservicebus/vscode.md

```diff
@@ -0,0 +1,263 @@
+
+## Prerequisities
+
+* This article assumes knowledge of NServiceBus solutions.
+* VSCode must have the [C# for Visual Studio (powered by OmniSharp)](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) extension installed.
+
+## VSCode configuration files
+
+The build and debug system in VSCode is controlled by two files located in a `.vscode` directory at the project root:
+
+* _{Project Root}_
+    * `.vscode/`
+        * `launch.json`
+        * `tasks.json`
+
+The **launch.json** file describes different projects within the solution, and describes how to launch each one.
+
+The **tasks.json** file describes actions to take, the most common of which is to build the code, which can be executed as a prerequisite of executing a configuration in **launch.json**.
+
+## tasks.json
+
+The simplest **tasks.json** file will describe how to build the solution using `dotnet build`:
+
+```json
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "build",
+            "command": "dotnet",
+            "type": "process",
+            "args": [
+                "build",
+                "${workspaceFolder}/path/to/SolutionName.sln"
+            ],
+            "problemMatcher": "$msCompile"
+        }
+    ]
+}
+```
+
+For an NServiceBus solution, use the content above and adjust the path of the `sln` file.
+
+For more information on **tasks.json** see [Visual Studio Code: Integrate with External Tools via Tasks](https://code.visualstudio.com/docs/editor/tasks).
+
+## launch.json
+
+The high-level structure of the launch.json file contains a collection of individual project objects in `configurations`, and then (optionally) a `compounds` collection that group up multiple configurations that will be launched together.
+
+```json
+{
+    "version": "0.2.0",
+    "configurations: [
+
+    ],
+    "compounds": [
+
+    ]
+}
+```
+
+### Console application
+
+Here is an example configuration for an NServiceBus endpoint hosted as a console application:
+
+```json
+{
+    "name": "Sales",
+    "type": "coreclr",
+    "request": "launch",
+    "preLaunchTask": "build",
+    "program": "${workspaceFolder}/path/to/Sales/bin/Debug/netcoreapp2.1/Sales.dll",
+    "args": [],
+    "cwd": "${workspaceFolder}/path/to/Sales",
+    "stopAtEntry": false,
+    "console": "externalTerminal"
+}
+```
+
+Most of the values can be considered boilerplate except for the `program` and `cwd` values which should be adjusted for each project:
+
+* `name`: Provides a name that appears for the project in VSCode's Debug dropdown.
+* `type`: The value `coreclr` identifies the plugin that should handle debugging responsibilities.
+* `request`: The value `launch` launches the process, rather than attaching to an existing process.
```

Comment:
The value `launch` means a new process will be started, rather than attaching to an existing process.